pyhap/hap_handler.py

```diff
@@ -596,60 +600,93 @@ def handle_set_characteristics(self):
     def handle_pairings(self):
         """Handles a client request to update or remove a pairing."""
         if not self.is_encrypted:
-            raise UnprivilegedRequestException
+            self._send_authentication_error_tlv_response(HAP_TLV_STATES.M2)
+            return
 
         tlv_objects = tlv.decode(self.request_body)
         request_type = tlv_objects[HAP_TLV_TAGS.REQUEST_TYPE][0]
         if request_type == 3:
             self._handle_add_pairing(tlv_objects)
         elif request_type == 4:
             self._handle_remove_pairing(tlv_objects)
+        elif request_type == 5:
+            self._handle_list_pairings()
         else:
             raise ValueError(
                 "Unknown pairing request type of %s during pair verify" % (request_type)
             )
 
     def _handle_add_pairing(self, tlv_objects):
         """Update client information."""
-        logger.debug("Adding client pairing.")
+        logger.debug("%s: Adding client pairing.", self.client_address)
         client_username = tlv_objects[HAP_TLV_TAGS.USERNAME]
         client_public = tlv_objects[HAP_TLV_TAGS.PUBLIC_KEY]
         client_uuid = uuid.UUID(str(client_username, "utf-8"))
         should_confirm = self.accessory_handler.pair(client_uuid, client_public)
         if not should_confirm:
-            self.send_response_with_status(
-                500, HAP_SERVER_STATUS.INVALID_VALUE_IN_REQUEST
-            )
+            self._send_authentication_error_tlv_response(HAP_TLV_STATES.M2)
             return
 
-        data = tlv.encode(HAP_TLV_TAGS.SEQUENCE_NUM, b"\x02")
-        self.send_response(200)
-        self.send_header("Content-Type", self.PAIRING_RESPONSE_TYPE)
-        self.end_response(data)
-
-        # Avoid updating the announcement until
-        # after the response is sent as homekit will
-        # drop the connection and fail to pair if it
-        # sees the accessory is now paired as it doesn't
-        # know that it was the one doing the pairing.
-        self.accessory_handler.finish_pair()
```

Comment:
The mdns announcement was being updated every time a new appletv/hub was added.  We only need to set the `sf` once when we finish `_pairing_five`